test/tap/tests/unit/plugin_servers_cluster_transport_unit-t.cpp

```
@@ -0,0 +1,547 @@
+	lock.unlock();
+	pthread_mutex_unlock(&pull_mutex);
+	worker.join();
+	return completed;
+}
+
+bool simultaneous_server_module_polls_finish_after_v2_pull(
+	ProxySQL_Node_Entry& node, MYSQL_RES* result, ProxySQL_Cluster& cluster) {
+	std::mutex done_mutex;
+	std::condition_variable done_cv;
+	bool done = false;
+	pthread_mutex_lock(&cluster.update_mysql_servers_v2_mutex);
```

Cppcheck (2.21.0)

[warning] If memory allocation fails, then there is a possible null pointer dereference

(nullPointerOutOfMemory)